src/engine/controls/bpmcontrol.cpp

```diff
@@ -763,21 +763,15 @@ double BpmControl::getBeatMatchPosition(
     double dOtherBeatFraction;
     // If not, we have to figure it out
     EngineBuffer* pOtherEngineBuffer = pickSyncTarget();
-    if (pOtherEngineBuffer == nullptr) {
-        if (playing) {
-            // Sync to itself if we are already playing
-            pOtherEngineBuffer = getEngineBuffer();
-        } else {
-            return dThisPosition;
-        }
-    }
-
     if (playing) {
-        // "this" track is playing, or just starting
-        // only match phase if the sync target is playing as well
-        if (pOtherEngineBuffer->getSpeed() == 0.0) {
-            return dThisPosition;
+        if (!pOtherEngineBuffer || pOtherEngineBuffer->getSpeed() == 0.0) {
+            // "this" track is playing, or just starting
+            // only match phase if the sync target is playing as well
+            // else use the previous phase of "this" track before the seek
+            pOtherEngineBuffer = getEngineBuffer();
```

Comment:
doesn't setting pOtherEngineBuffer to ourselves mean there is no seek? how is this different from returning dThisPosition?

Reply:
dThisPosition returns the unchanged target position. If we sync to ourselfes it syncs to the position the deck was before the seek.